tasks/main.yml

```
@@ -9,6 +9,9 @@
   ansible.builtin.package:
     name: systemd-resolved
     state: present
+  when:
+    - ansible_distribution | lower == 'debian'
+    - ansible_distribution_major_version >= '12'
```

We need to use ansible_facts to allow disabling fact variable injection.
```suggestion
    - ansible_facts.distribution | lower == 'debian'
    - ansible_facts.distribution_major_version >= '12'
```